packages/fx-core/src/plugins/resource/bot/plugin.ts

```diff
@@ -174,20 +176,23 @@ export class TeamsBotImpl {
       ConfigNames.PROGRAMMING_LANGUAGE,
       this.config.scaffold.programmingLanguage
     );
-    // CheckThrowSomethingMissing(ConfigNames.GRAPH_TOKEN, this.config.scaffold.graphToken);
-    CheckThrowSomethingMissing(ConfigNames.SUBSCRIPTION_ID, this.config.provision.subscriptionId);
-    CheckThrowSomethingMissing(ConfigNames.RESOURCE_GROUP, this.config.provision.resourceGroup);
-    CheckThrowSomethingMissing(ConfigNames.LOCATION, this.config.provision.location);
-    CheckThrowSomethingMissing(ConfigNames.SKU_NAME, this.config.provision.skuName);
-    CheckThrowSomethingMissing(CommonStrings.SHORT_APP_NAME, this.ctx.projectSettings?.appName);
 
-    if (!this.config.provision.siteName) {
-      this.config.provision.siteName = ResourceNameFactory.createCommonName(
-        this.config.resourceNameSuffix,
-        this.ctx.projectSettings?.appName,
-        MaxLengths.WEB_APP_SITE_NAME
-      );
-      Logger.debug(`Site name generated to use is ${this.config.provision.siteName}.`);
+    if (!isArmSupportEnabled()) {
```

Comment:
Nit: If the if( ! isArmSupportEnabled()) is the main logic, it's better to use if(isArmSupportEnabled()) when needed.

Reply:
The scenario is that when provisioning bot plugin azure resources. 
1. arm support enabled: 
    - provision bot aad app.
2. no arm support enabled:
    - provision bot aad app.
    - provision azure resources (bot service, web app, etc.)
    
 Since both scenario need to do the "provision bot aad app" step and only non-arm situation needs to provision azure resources, we have the `if( ! isArmSupportEnabled())` to see whether provision azure resources here.